ams/AssetAnalyzer.cs

```diff
@@ -139,13 +140,19 @@ public override async Task MigrateAsync(CancellationToken cancellationToken)
             await MigrateInParallel(assets, filteredList, async (asset, cancellationToken) =>
             {
                 var storage = await _resourceProvider.GetStorageAccountAsync(account, asset, cancellationToken);
-
-                var result = await AnalyzeAsync(asset, storage, cancellationToken);
-                var assetType = result.AssetType ?? "unknown";
-                assetTypes.AddOrUpdate(assetType, 1, (key, value) => Interlocked.Increment(ref value));
-                reportGenerator?.WriteRow(result);
-                statistics.Update(result);
-                await writer.WriteAsync(statistics.Total, cancellationToken);
+                if (storage != null)
```

Comment:
Is there any special case that the storage account is not available for a valid asset? it seems no need to specially handle it, just keeping the original code logic seems fine.

Reply:
the same applies to all other classes with the similar changes.